indico_toolkit/indico_wrapper/workflow.py

```diff
@@ -94,50 +94,43 @@ def get_submission_object(self, submission_id: int) -> Submission:
     def get_submission_results_from_ids(
         self,
         submission_ids: List[int],
-        timeout: int = 75,
-        ignore_exceptions: bool = False,
+        timeout: int = 120,
+        return_raw_json: bool = False,
+        raise_exception_for_failed: bool = True,
+        return_failed_results: bool = True,
     ) -> List[WorkflowResult]:
         """
         Wait for submission to pass through workflow models and get result. If Review is enabled, result may be retrieved prior to human review.
         Args:
             submission_id (int): Id of submission predictions to retrieve
             timeout (int): seconds permitted for each submission prior to timing out
-            ignore_exceptions (bool): if True, catch exception for unsuccessful submission
-
+            return_raw_json: (bool) = If True return raw json result, otherwise return WorkflowResult object.
+            raise_exception_for_failed (bool): if True, ToolkitStatusError raised for failed submissions
+            return_failed_results (bool): if True, return objects for failed submissions
         Returns:
             List[WorkflowResult]: workflow result objects
         """
         results = []
         for subid in submission_ids:
-            try:
-                job = self.client.call(
-                    SubmissionResult(subid, wait=True, timeout=timeout)
-                )
-            except IndicoRequestError as e:
-                message = f"IndicoRequestError with Submission {subid}: {e}"
-                self._error_handle(message, ignore_exceptions)
-                continue
-            if job.status != "SUCCESS":
-                message = f"{job.status}! Submission {subid}: {job.result}"
-                self._error_handle(message, ignore_exceptions)
-                continue
-            results.append(WorkflowResult(self.get_storage_object(job.result)))
+            submission: Submission = self.wait_for_submission(subid, timeout=timeout)
+            if submission.status == "FAILED":
+                message = f"FAILURE, Submission: {subid}. {submission.errors}"
+                if raise_exception_for_failed:
+                    raise ToolkitStatusError(message)
+                elif not return_failed_results:
+                    print(message)
```

Comment:
it will be helpful to print the submission id here too

Reply:
'subid' is in here both places